File: source/BolaoInfra/Models/Grupo.cs
using System;
using System.Collections.Generic;

namespace BolaoInfra.Models
{
    public partial class Grupo
    {
        public Grupo()
        {
            Apostadores = new HashSet<Apostador>();
        }

        public int CodGrupo { get; set; }
        public string Nome { get; set; }
        public int CodApostResponsavel { get; set; }

        public virtual Apostador ApostadorResponsavel { get; set; }

        public virtual ICollection<Apostador> Apostadores { get; set; }
    }
}

File: source/bolao_app/lib/models/grupo.dart
class Grupo {
  int codGrupo;
  String nome;
  int codApostResponsavel;

  Grupo({required this.codGrupo,
    required this.nome,
    required this.codApostResponsavel
  });

  factory Grupo.fromJson(dynamic json) {
    return Grupo(codGrupo: json['codGrupo'] as int,
        nome: json['nome'] as String,
        codApostResponsavel: json['codApostResponsavel'] as int);
  }
}